### PR TITLE
search docs: reflect default Standard search in query syntax

### PR DESCRIPTION
**Pull request by rvantonder**
As in title. I will update https://docs.sourcegraph.com/code_search/reference/language#search-pattern in a separate PR, this is ready to ship.

Screen Shot 2022-09-02 at 2 11 26 PM

Link to current docs here: https://docs.sourcegraph.com/code_search/reference/queries

## Test plan
Doc update

### CODE REVIEW

File: doc/code_search/reference/queries.md [hunk 1]
```diff
@@ -38,25 +38,39 @@ This page describes search pattern syntax and keywords available for code search
...
+| [`/foo.*bar/`](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+/foo.*bar/&patternType=standard) | Match the regular expression `foo.*bar`.                      We support [RE2 syntax](https://golang.org/s/re2syntax).                                                        |
+| [`foo` `AND` `bar`](https://sourcegraph.com/search?q=context:global+foo+AND+bar&patternType=standard) | Match documents containing both `foo` _and_ `bar` anywhere in the document. |
 
-### Regular expression search
+Matching is case-_insensitive_ (toggle the <span class="toggle-container"><img class="toggle" src=../img/case.png alt="case"></span> button to change).
 
-Click the <span class="toggle-container"><img class="toggle" src=../img/regex.png alt="regular expression"></span> toggle to interpret search patterns as regexps. [RE2 syntax](https://golang.org/s/re2syntax) is supported. In general, special characters may be escaped with `\`. Here is a list of valid syntax and behavior:
+### Dedicated regular expression search
```
Comment (rvantonder) [inline doc/code_search/reference/queries.md:54]: I'd really like to delete this :-3. One day.

Comment (rvantonder) [reply doc/code_search/reference/queries.md:54]: Actually, I wonder if we could make this collapsible... Put it in a 

```
<details>
  <summary>Dedicated regular expression search with toggle</summary>
</details>
```

Comment (rvantonder) [reply doc/code_search/reference/queries.md:54]: ehm, PR for feedback on that: https://github.com/sourcegraph/sourcegraph/pull/41290

**Review by tbliu98 — APPROVED** [review]

File: doc/code_search/reference/queries.md [hunk 2]
```diff
@@ -38,25 +38,39 @@ This page describes search pattern syntax and keywords available for code search
...
-| [`"foo bar"`](https://sourcegraph.com/search?q=%22foo+bar%22&patternType=literal) | Match the string `"foo bar"`. The quotes are matched literally.                                                                                                                                                                       |
+| [`foo bar`](https://sourcegraph.com/search?q=foo+bar&patternType=standard)         | Match the string `foo bar` exactly. No need to add quotes, we match `foo` followed by `bar`, with exactly one space between the terms. |
+| [`"foo bar"`](https://sourcegraph.com/search?q=%22foo+bar%22&patternType=standard) | Match the string `"foo bar"` exactly. The quotes are matched literally.                                                                                                                                                                       |
+| [`/foo.*bar/`](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+/foo.*bar/&patternType=standard) | Match the regular expression `foo.*bar`.                      We support [RE2 syntax](https://golang.org/s/re2syntax).                                                        |
```
Comment (tbliu98) [inline doc/code_search/reference/queries.md:49]: Maybe better to **bold** the words "regular expression" in "Match the regular expression..." if we decide to make the dedicated regex search section of this doc collapsible? So it's super clear to readers looking for "how to do a regex search" that `/.../` will serve their need

Comment (rvantonder) [reply doc/code_search/reference/queries.md:49]: yep, easy and agree

**Review by camdencheek — APPROVED** [review]

**Pull request merged**